sdks/java/io/contextual-text-io/src/main/java/org/apache/beam/sdk/io/contextualtextio/RecordWithMetadata.java

```diff
@@ -17,69 +17,103 @@
  */
 package org.apache.beam.sdk.io.contextualtextio;
 
-import com.google.auto.value.AutoValue;
-import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.io.FileSystems;
 import org.apache.beam.sdk.io.fs.ResourceId;
-import org.apache.beam.sdk.schemas.AutoValueSchema;
-import org.apache.beam.sdk.schemas.annotations.DefaultSchema;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.Schema.FieldType;
+import org.apache.beam.sdk.schemas.Schema.LogicalType;
+import org.apache.beam.sdk.values.Row;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions;
 
 /**
- * Helper Class based on {@link AutoValueSchema}, it provides Metadata associated with each Record
- * when reading from file(s) using {@link ContextualTextIO}.
+ * Helper Class based on {@link Row}, it provides Metadata associated with each Record when reading
+ * from file(s) using {@link ContextualTextIO}.
  *
  * <h3>Fields:</h3>
  *
  * <ul>
  *   <li>recordOffset: The offset of a record (the byte at which the record begins) in a file. This
  *       information can be useful if you wish to reconstruct the file. {@link
- *       RecordWithMetadata#getRecordOffset()}
+ *       RecordWithMetadata#RANGE_OFFSET}
  *   <li>recordNum: The ordinal number of the record in its file. {@link
- *       RecordWithMetadata#getRecordNum()}
- *   <li>recordValue: The value / contents of the record {@link RecordWithMetadata#getValue()}
+ *       RecordWithMetadata#RECORD_NUM}
+ *   <li>recordValue: The value / contents of the record {@link RecordWithMetadata#VALUE}
  *   <li>rangeOffset: The starting offset of the range (split), which contained the record, when the
- *       record was read. {@link RecordWithMetadata#getRangeOffset()}
+ *       record was read. {@link RecordWithMetadata#RANGE_OFFSET}
```

Comment:
@lukecwik I think these are still useful to external , as we use them to create the pos information, but others may want to do their own implementations as we do have a option to disable the shuffle step we use, so this would then allow folks to roll their own uses.

Reply:
good point